pkg/types/hashedrekord/v0.0.1/entry.go

```diff
@@ -178,11 +179,21 @@ func (v *V001Entry) validate() (pki.Signature, pki.PublicKey, error) {
 		return nil, nil, types.ValidationError(errors.New("invalid value for hash"))
 	}
 
+	var alg crypto.Hash
+	switch swag.StringValue(hash.Algorithm) {
+	case models.HashedrekordV001SchemaDataHashAlgorithmSha384:
+		alg = crypto.SHA384
+	case models.HashedrekordV001SchemaDataHashAlgorithmSha512:
+		alg = crypto.SHA512
+	default:
+		alg = crypto.SHA256
+	}
+
 	decoded, err := hex.DecodeString(*hash.Value)
 	if err != nil {
 		return nil, nil, err
 	}
-	if err := sigObj.Verify(nil, keyObj, options.WithDigest(decoded)); err != nil {
+	if err := sigObj.Verify(nil, keyObj, options.WithDigest(decoded), options.WithCryptoSignerOpts(alg)); err != nil {
```

Comment:
nit for another file: Should we add a comment that the digest in https://github.com/sigstore/rekor/blob/fc28ac1bc2ecae172371df6b63efe48daee863ff/pkg/pki/x509/x509.go#L87 is a default? I was looking through the code and was initially uncertain if that needed to be modified too.

Reply:
I think I was missing this in my PR #1945 .